coalib/misc/Shell.py

```diff
@@ -102,7 +140,7 @@ def run_shell_command(command, stdin=None, **kwargs):
     :return:        A tuple with ``(stdoutstring, stderrstring)``.
     """
     with run_interactive_shell_command(command, **kwargs) as p:
-        ret = p.communicate(stdin)
+        ret = ShellCommandResult(p, stdin)
```

Comment:
you think that makes sense to move the functionality into the result holding class? not sure...